src/main/java/ch/njol/skript/lang/SkriptParser.java

```diff
@@ -261,6 +261,18 @@ private <T extends SyntaxElement> T parse(Iterator<? extends SyntaxElementInfo<?
 								x = x2;
 							}
 							T t = info.c.newInstance();
+							if (getParser().getCurrentStructure() != null) {
+								List<TriggerSection> sections = getParser().getCurrentSections();
+								TriggerSection current = sections == null || sections.isEmpty() ? null : sections.get(sections.size() - 1);
+								if (t.getUsableStructures().length > 0 && !CollectionUtils.contains(t.getUsableStructures(), getParser().getCurrentStructure().getClass())) {
+									Skript.error("You cannot use this element in " + Utils.a(getParser().getCurrentStructure().toString(null, false)) + " structure.");
+								} else if (t.getUsableSections().length > 0 && (current == null || !CollectionUtils.contains(t.getUsableSections(), current.getClass()))) {
+									Skript.error("You cannot use this element " + (current == null ? "outside of the allowed sections."
+										: "in a '" + current.toString(null, false) + "' section"));
+								}
+							} else if (t.getUsableStructures().length > 0) {
+								Skript.error("You cannot use this element outside of the allowed structures.");
```

Comment:
Applies to all the other errors, is `this element` clear which element it's referring to? if not then it should be included in the error message

In the error it's probably referring to the line where the structure is but I guess that isn't always very clear to all users

Reply:
I've tested it before and from what I remember (i'll retest in a bit) it's just like any other message and outputs the line number, skript and error line and element as any other do, but ill double check a bit later.